datafusion-jni/src/dataframe.rs

```diff
@@ -24,11 +24,27 @@ pub extern "system" fn Java_org_apache_arrow_datafusion_DataFrames_collectDatafr
     let dataframe = unsafe { &*(dataframe as *const DataFrame) };
     let schema = dataframe.schema().into();
     runtime.block_on(async {
-        let batches = dataframe
-            .clone()
-            .collect()
-            .await
-            .expect("failed to collect dataframe");
+        let batches = dataframe.clone().collect().await;
+        let batches = match batches {
```

Comment:
Slightly unrelated to the multiple paths feature, but when testing this I found it was easily possible to get a panic from rust here as eg. a syntax error in a query isn't raised until collecting the dataframe.